shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,194 @@
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        packaging = self.env["product.packaging"].sudo().browse(packaging_id)
+        message = None
+        next_packaging = None
+        if not packaging:
+            message = self.msg_store.record_not_found()
+        elif not cancel and self._check_dimension_to_update(kwargs):
+            self._update_packaging_dimension(packaging, kwargs)
+            message = self.msg_store.packaging_dimension_updated(packaging)
+        if packaging:
+            next_packaging = self._get_next_packaging_to_set_dimension(
+                selected_line.product_id, packaging
+            )
+        if not next_packaging:
+            return self._response_for_set_quantity(
+                picking, selected_line, message=message
+            )
+        return self._response_for_set_packaging_dimension(
+            picking, selected_line, next_packaging, message=message
+        )
+
+    def _check_dimension_to_update(self, dimensions):
+        """Return True if there is any dimension that needs to be updated on the packaging."""
+        return any([value is not None for key, value in dimensions.items()])
+
+    def _get_dimension_fields_conversion_map(self):
+        return {"weight": "max_weight", "length": "packaging_length"}
+
+    def _update_packaging_dimension(self, packaging, dimensions_to_update):
+        """Update dimension on the packaging."""
+        fields_conv_map = self._get_dimension_fields_conversion_map()
+        for dimension, value in dimensions_to_update.items():
+            if value is not None:
+                dimension = fields_conv_map.get(dimension, dimension)
+                packaging[dimension] = value
+
+
+class ShopfloorReceptionValidator(Component):
+    _inherit = "shopfloor.reception.validator"
+
+    def set_packaging_dimension(self):
+        return {
+            "picking_id": {"coerce": to_int, "required": True, "type": "integer"},
+            "selected_line_id": {
+                "coerce": to_int,
+                "required": True,
+                "type": "integer",
+            },
+            "packaging_id": {"coerce": to_int, "required": True, "type": "integer"},
+            "height": {
+                "coerce": to_float,
+                "required": False,
+                "type": "float",
+                "nullable": True,
+            },
+            "length": {
+                "coerce": to_float,
+                "required": False,
+                "type": "float",
+                "nullable": True,
+            },
+            "width": {
+                "coerce": to_float,
+                "required": False,
+                "type": "float",
+                "nullable": True,
+            },
+            "weight": {
+                "coerce": to_float,
+                "required": False,
+                "type": "float",
+                "nullable": True,
+            },
```

Comment:
We have to check the request for max_weight, because this will be send from javascript.
```suggestion
            "max_weight": {
                "coerce": to_float,
                "required": False,
                "type": "float",
                "nullable": True,
            },
```

Reply:
max_weight is another field, IIRC.
If I'm not mistaking, we should rather fix the frontend.

Reply:
Yes there is weight defined in OCA/product-attribute/product_packaging_dimension
But this is not a dependency of this module. 
In odoo standard there is only max_weight

Reply:
also the packaging parser is getting the data from max_weight and putting into in js also in max_weight.
So i should be max_weight here

Reply:
Thank you, updated